app/services/container_dashboard_service.rb

```diff
@@ -73,51 +73,34 @@ def status
   end
 
   def providers
+    provider_classes_to_ui_types = {
+      "ManageIQ::Providers::Openshift::ContainerManager"           => :openshift,
+      "ManageIQ::Providers::OpenshiftEnterprise::ContainerManager" => :openshift,
+      "ManageIQ::Providers::Kubernetes::ContainerManager"          => :kubernetes,
+      "ManageIQ::Providers::Atomic::ContainerManager"              => :atomic,
+      "ManageIQ::Providers::AtomicEnterprise::ContainerManager"    => :atomic
+    }
+
     if @ems.present?
-      if @ems.kind_of?(ManageIQ::Providers::Openshift::ContainerManager) ||
-         @ems.kind_of?(ManageIQ::Providers::OpenshiftEnterprise::ContainerManager)
-        {
-          :iconClass    => "pficon pficon-openshift",
-          :id           => "openshift",
-          :providerType => "OpenShift"
-        }
-      elsif @ems.kind_of?(ManageIQ::Providers::Kubernetes::ContainerManager)
-        {
-          :iconClass    => "pficon pficon-kubernetes",
-          :id           => "kubernetes",
-          :providerType => "Kubernetes"
-        }
-      elsif @ems.kind_of?(ManageIQ::Providers::Atomic::ContainerManager) ||
-            @ems.kind_of?(ManageIQ::Providers::AtomicEnterprise::ContainerManager)
-        {
-          :iconClass    => "pficon icon-atomic",
-          :id           => "atomic",
-          :providerType => "Atomic"
-        }
-      end
+      ui_type = provider_classes_to_ui_types[@ems.type]
```

Comment:
I'm surely missing something, but this code is repeated below around lines 95-99